pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -568,6 +568,29 @@ func (s *ServiceRuntime) SetDesiredKubeObserveObject(obj client.Object, objectNa
 	return s.SetDesiredComposedResourceWithName(kobj, objectName)
 }
 
+// SetDesiredKubeObserveObjectWithLabels takes any `runtime.Object`, puts it into a provider-kubernetes Object and then
+// adds it to the desired composed resources.
+func (s *ServiceRuntime) SetDesiredKubeObserveObjectWithLabels(obj client.Object, objectName string, labels map[string]string, refs ...xkube.Reference) error {
```

Comment:
We can use functional parameters here and simply extend `SetDesiredKubeObserveObject()`. Take a look at the `SetDesiredKubeObject` function, especially the `opts ...KubeObjectOption` part.

We can even get rid of `SetDesiredKubeObserveObject` completely by having a `KubeObjectOptionObserve` function.

Reply:
We did that together, resolving then